### PR TITLE
Client-Server protocol stabilisation 

### PR DESCRIPTION
**Pull request by emnigma**
- [x] Update and test client-server communication: Update protocol & structure
- [ ] Solve mismatch between old and new model input shapes (will be done in different PR)
- [ ] Solve mismatch between old and new map names (will be done in different PR)

### CODE REVIEW

File: AIAgent/launch_servers.py [hunk 1]
```diff
@@ -86,7 +86,10 @@ def run_server_instance(port: int, start_server: bool) -> ServerInstanceInfo:
     launch_server = [
         "dotnet",
         "VSharp.ML.GameServer.Runner.dll",
-        "--checkactualcoverage",
+        "--datasetdescription",
```
Comment (gsvgit) [inline AIAgent/launch_servers.py:89]: This argument is not required in server mode.

File: AIAgent/config.py [hunk 2]
```diff
@@ -19,6 +19,7 @@ class GeneralConfig:
     IMPORT_MODEL_INIT = ...
     EXPORT_MODEL_INIT = ...
     DEVICE = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+    DATASET_BASE_PATH = "/Users/emax/Data/PySymGym/maps/DotNet/Maps/ManuallyCollected/bin/Debug/net7.0"
```
Comment (gsvgit) [inline AIAgent/config.py:22]: Hardcoded absolute path.

File: AIAgent/common/constants.py [hunk 3]
```diff
@@ -47,4 +47,6 @@ class ResultsHandlerLinks:
 BASE_NN_OUT_FEATURES_NUM = 8
 
 # assuming we start from /VSharp/VSharp.ML.AIAgent
-SERVER_WORKING_DIR = "../VSharp.ML.GameServer.Runner/bin/Release/net7.0/"
+SERVER_WORKING_DIR = (
+    "/Users/emax/Data/PySymGym/GameServers/VSharp/VSharp.ML.GameServer.Runner/bin/Debug/net7.0/"
```
Comment (gsvgit) [inline AIAgent/common/constants.py:51]: Hardcoded absolute path.

File: AIAgent/launch_servers.py (hunk 1 above)
Comment (emnigma) [reply AIAgent/launch_servers.py:89]: Without this argument server doesn't launch:
```
Unhandled exception. System.ArgumentException: The value cannot be an empty string. (Parameter 'path')
   at System.ArgumentException.ThrowNullOrEmptyException(String argument, String paramName)
   at System.IO.File.ReadAllText(String path, Encoding encoding)
   at Main.loadGameMaps(String datasetDescriptionFilePath) in /Users/emax/Data/PySymGym/GameServers/VSharp/VSharp.ML.GameServer.Runner/Main.fs:line 42
   at Main.main(String[] args) in /Users/emax/Data/PySymGym/GameServers/VSharp/VSharp.ML.GameServer.Runner/Main.fs:line 252
```
moreover, this path should be valid:
```
Unhandled exception. System.IO.DirectoryNotFoundException: Could not find a part of the path '/Users/emax/Data/PySymGym/GameServers/VSharp/VSharp.ML.GameServer.Runner/bin/Debug/net7.0/maps/DotNet/Maps/dataset.json'.
   at Interop.ThrowExceptionForIoErrno(ErrorInfo errorInfo, String path, Boolean isDirError)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String path, OpenFlags flags, Int32 mode, Func`4 createOpenException)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize, UnixFileMode openPermissions, Int64& fileLength, UnixFileMode& filePermissions, Func`4 createOpenException)
   at System.IO.Strategies.OSFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize, Nullable`1 unixCreateMode)
   at System.IO.StreamReader.ValidateArgsAndOpenPath(String path, Encoding encoding, Int32 bufferSize)
   at System.IO.File.ReadAllText(String path, Encoding encoding)
   at Main.loadGameMaps(String datasetDescriptionFilePath) in /Users/emax/Data/PySymGym/GameServers/VSharp/VSharp.ML.GameServer.Runner/Main.fs:line 42
   at Main.main(String[] args) in /Users/emax/Data/PySymGym/GameServers/VSharp/VSharp.ML.GameServer.Runner/Main.fs:line 252
```

Comment (emnigma) [reply AIAgent/launch_servers.py]: This option is removed in latest commit

File: AIAgent/config.py (hunk 2 above)
Comment (emnigma) [reply AIAgent/config.py]: Fixed

File: AIAgent/common/constants.py (hunk 3 above)
Comment (emnigma) [reply AIAgent/common/constants.py]: Fixed

**Pull request merged**